chessboard/gallen/chessboard.go

```diff
@@ -0,0 +1,45 @@
+package chessboard
+
+const RankSize = 8
+
+type Rank [RankSize]bool
+
+type Chessboard map[string]Rank
+
+func CountInRank(cb Chessboard, rank string) (count int) {
+    for _, occupied := range cb[rank] {
+        if (occupied) {
+            count++
+        }
+    }
+    return
+}
+
+func CountInFile(cb Chessboard, file int) (count int) {
+    if (file <= RankSize) {
+        for _, rank := range cb {
+            if (rank[file-1]) {
+                count++
+            }
+        }
+    }
+    return
+}
+
+func CountAll(cb Chessboard) int {
+    for _, rank := range cb {
+        return len(cb) * len(rank)
```

Comment:
Noooo Greg! You failed me here! You have to use your const here. The whole function becomes:
```go
return len(cb) * RankSize
```

Reply:
Yeah, you are correct. The problem is I hate the hard-coding of the size.  And even worse we can only hard-code the rank.

Reply:
I think in this case board both dimensions should be hard coded but... :man_shrugging: